File: JavaModule/src/main/java/com/example/lee/thirdRound/ReverseWordsinaString.java
package com.example.lee.thirdRound;

public class ReverseWordsinaString {
    public String reverseWords(String s) {
        if (s == null || s.length() < 1) {
            return s;
        }

        String[] fields = s.trim().split(" +", -1);
        int l = 0, r = fields.length - 1;
        while (l < r) {
            String t = fields[l];
            fields[l] = fields[r];
            fields[r] = t;
            l++;
            r--;
        }
        return String.join(" ", fields);
    }
}

File: python_module/examples/reverse_words_in_a_string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        if not s:
            return s
        slist = [x for x in s.split(" ") if x]
        l, r = 0, len(slist) - 1
        while l < r:
            slist[l], slist[r] = slist[r], slist[l]
            l += 1
            r -= 1

        return " ".join(slist)